vector/src/debug/java/im/vector/app/features/debug/features/DebugVectorFeatures.kt

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) 2021 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package im.vector.app.features.debug.features
+
+import android.content.Context
+import android.content.SharedPreferences
+import im.vector.app.features.DefaultVectorFeatures
+import im.vector.app.features.VectorFeatures
+import kotlin.reflect.KClass
+
+class DebugVectorFeatures(
+        context: Context,
+        private val vectorFeatures: DefaultVectorFeatures
+) : VectorFeatures {
+
+    private val featurePrefs = context.getSharedPreferences("debug-features", Context.MODE_PRIVATE)
```

Comment:
We use `DataStore` nowadays, even if I did not find an easy way to just read a value synchronously on it.

Reply:
will update :+1:

Reply:
https://github.com/vector-im/element-android/pull/4635/commits/05ce1414b973422f5e65bb937aed665ff934aef0 using the datastore api with `runBlocking` as the suspending is only needed for debug builds